contracts/oracles/WamplOracle.sol

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.4;
+
+import "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import {IOracle} from "../interfaces/IOracle.sol";
+import {IWAMPL} from "../interfaces/IWAMPL.sol";
+import {IChainlinkAggregator} from "../interfaces/IChainlinkAggregator.sol";
+
+/**
+ * @title WamplOracle Oracle
+ *
+ * @notice Provides a value onchain from a chainlink oracle aggregator
+ */
+contract WamplOracle is IOracle {
+    /// @dev The output price has a 8 decimal point precision.
+    uint256 public constant PRICE_DECIMALS = 8;
+    // The address of the Chainlink Aggregator contract
+    IChainlinkAggregator public immutable amplEthOracle;
+    IChainlinkAggregator public immutable ethUsdOracle;
+    IWAMPL public immutable wampl;
+    uint256 public immutable stalenessThresholdSecs;
+    uint256 public immutable amplDecimals;
+    uint256 public immutable wamplDecimals;
+    int256 public immutable convertPriceByDecimals;
+
+    constructor(
+        IChainlinkAggregator _amplEthOracle,
+        IChainlinkAggregator _ethUsdOracle,
+        IWAMPL _wampl,
+        uint256 _stalenessThresholdSecs
+    ) {
+        amplEthOracle = _amplEthOracle;
+        ethUsdOracle = _ethUsdOracle;
+        wampl = _wampl;
+        stalenessThresholdSecs = _stalenessThresholdSecs;
+        amplDecimals = uint256(IERC20Metadata(_wampl.underlying()).decimals());
+        wamplDecimals = uint256(_wampl.decimals());
+        convertPriceByDecimals =
+            int256(uint256(_amplEthOracle.decimals())) +
```

Comment:
Casting from uint8 to int256 results in a compiler error because it's ambiguous how the conversion should be handled for negative numbers. The double cast here resolves this ambiguity.

```
uint8 val = 255; // (2**8)-1
uint256 val2 = uint256(val);
// val2 = 255
int256 val3 = int256(val2);
// val3 = 255
```
vs
```
uint8 val = 255; // (2**8)-1
int8val2 = int8(val);
// val2 = -1
int256 val3 = int256(val2);
// val3 = -1
```